File: leetcode/035-search-insert-position.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        l, cnt = 0, len(nums)
        while cnt > 0:
            step = cnt / 2
            m = l + step
            if nums[m] < target:
                l = m + 1
                cnt -= step + 1
            else:
                cnt = step
        return l

File: leetcode/035-search-insert-position.cpp
class Solution {
 public:
  int searchInsert(vector<int>& nums, int target) {
    int l = 0;
    for (int cnt = nums.size(); cnt > 0; ) {
      int step = cnt / 2;
      int m = l + step;
      if (nums[m] < target) {
        l = m + 1;
        cnt -= step + 1;
      } else cnt = step;
    }
    return l;
  }
};
